### ISSUE TITLE
Add readthedocs webhook

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#1565]
I got this in the e-mails and thought I should share.

>  Hello!
--
Your project, FakeItEasy, is currently using GitHub Services to trigger builds on Read the Docs. Effective January 31, 2019, GitHub will no longer process requests using the Services feature, and so Read the Docs will not receive notifications on updates to your repository.To continue building your Read the Docs project on changes to your repository, you will need to add a new webhook on your GitHub repository. You can either connect your GitHub account and configure a GitHub webhook integration, or you can add a generic webhook integration.You can find more information on our webhook intergrations in our documentation, at:https://docs.readthedocs.io/en/latest/webhooks.html#webhook-github-services

Hmm. I should probably also see if the e-mail could go to the fakeiteasy account. Oh. Unless you got the e-mail too, @thomaslevesque?

**Comment by blairconrad** [comment]
Added https://github.com/FakeItEasy/FakeItEasy/settings/hooks/79351154. We may not know if it works until we release or change the docs.

**Comment by blairconrad** [comment]
@thomaslevesque, I'm leaving open for a bit in case you want to check my settings. Otherwise, we can close and just keep an eye out for errors.

**Comment by thomaslevesque** [comment]
> Unless you got the e-mail too, @thomaslevesque?

I did

> I'm leaving open for a bit in case you want to check my settings

Sounds good to me. Thanks for taking care of this!

**Comment by thomaslevesque** [comment]
> > I'm leaving open for a bit in case you want to check my settings
> 
> Sounds good to me.

Sorry, I misread. I thought you wanted to keep open until we confirmed that it worked. The settings look good to me.

We could create a dummy branch to check that it builds correctly.

**Comment by blairconrad** [comment]
Good idea! Pushed https://github.com/FakeItEasy/FakeItEasy/tree/test-read-the-docs

**Comment by thomaslevesque** [comment]
Apparently it triggered a build, but I don't know if specifically checked out this branch. Apparently the build checks out a commit, not a branch.

**Comment by blairconrad** [comment]
Yeah. I saw that and wasn't sure what it built. The webhook was delivered successfully. I'm going to put a commit in there…

**Comment by blairconrad** [comment]
Okay. After enabling that branch in read the docs, and disabling the GitHub service, _then_ pushing a commit, I saw a successful build: https://fakeiteasy.readthedocs.io/en/test-read-the-docs/

I'm content if you are!

**Comment by blairconrad** [comment]
> Apparently it triggered a build, but I don't know if specifically checked out this branch. Apparently the build checks out a commit, not a branch.

It looked like it checked out a branch to me, if the logs are to be believed:

https://readthedocs.org/projects/fakeiteasy/builds/8411976/

```
git remote set-url origin https://github.com/FakeItEasy/FakeItEasy
git fetch --tags --prune --prune-tags --depth 50
git checkout --force origin/test-read-the-docs
git clean -d -f -f
python2.7 -mvirtualenv --no-site-packages --no-download /home/docs/checkouts/readthedocs.org/user_builds/fakeiteasy/envs/test-read-the-docs
/home/docs/checkouts/readthedocs.org/user_builds/fakeiteasy/envs/test-read-the-docs/bin/python -m pip install --upgrade --cache-dir /home/docs/checkouts/readthedocs.org/user_builds/fakeiteasy/.cache/pip pip
/home/docs/checkouts/readthedocs.org/user_builds/fakeiteasy/envs/test-read-the-docs/bin/python -m pip install --upgrade --cache-dir /home/docs/checkouts/readthedocs.org/user_builds/fakeiteasy/.cache/pip Pygments==2.2.0 setuptools<40 docutils==0.13.1 mock==1.0.1 pillow==2.6.1 alabaster>=0.7,<0.8,!=0.7.5 commonmark==0.5.4 recommonmark==0.4.0 mkdocs==0.17.3
cat mkdocs.yml
python /home/docs/checkouts/readthedocs.org/user_builds/fakeiteasy/envs/test-read-the-docs/bin/mkdocs build --clean
```

**Comment by thomaslevesque** [comment]
> I'm content if you are!

I am!

> It looked like it checked out a branch to me, if the logs are to be believed

Not in this one: https://readthedocs.org/projects/fakeiteasy/builds/8411899/

**Comment by blairconrad** [comment]
Oh. Um. That's odd. Maybe because it was started by the GitHub Service and not the webhook? (I am obviously making things up here.)

Regardless, I'll remove the branch and the Service integration, leaving us with only the webhook.

**Comment by blairconrad** [comment]
This change has been released as part of [FakeItEasy 5.1.0](https://github.com/FakeItEasy/FakeItEasy/releases/tag/5.1.0).
